src/Init/Data/Option/Lemmas.lean

```diff
@@ -677,6 +770,29 @@ theorem pmap_map (o : Option α) (f : α → β) {p : β → Prop} (g : ∀ b, p
       pmap (fun a h => g (f a) h) o (fun a m => H (f a) (mem_map_of_mem f m)) := by
   cases o <;> simp
 
+theorem pmap_congr.translate {α : Type u}
+    {p p' : α → Prop} (hp : ∀ x, p x ↔ p' x)
+    {o o' : Option α} (ho : o = o')
+    (h : ∀ x, x ∈ o → p x) : ∀ x, x ∈ o' → p' x := by
+  intro y hy
+  cases ho
+  exact (hp y).mp (h y hy)
+
+@[congr]
+theorem pmap_congr {α : Type u} {β : Type v}
+    {p p' : α → Prop} (hp : ∀ x, p x ↔ p' x)
+    {f : (x : α) → p x → β} {f' : (x : α) → p' x → β}
+    (hf : ∀ x h, f x ((hp x).mpr h) = f' x h)
+    {o o' : Option α} (ho : o = o')
+    {h : ∀ x, x ∈ o → p x} :
+    Option.pmap f o h = Option.pmap f' o' (Option.pmap_congr.translate hp ho h) := by
+  cases ho
+  unfold Option.pmap
```

Comment:
`unfold` is almost always a code smell because it suggests that basic API lemmas are missing, but in this case it turns out that it is not needed at all.
```suggestion
```